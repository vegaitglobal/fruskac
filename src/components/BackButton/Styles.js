/**
 * @flow
 */
import { StyleSheet, Platform } from "react-native";
import { tintColor } from "../../styles/CommonStyles";

export default StyleSheet.create({
  buttonHolder: {
    marginLeft: 8,
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  icon: {
    color: tintColor,
    marginRight: 8,
    top: Platform.OS === "ios" ? 2 : 1
  },
  text: { color: tintColor, fontSize: 16 }
});
