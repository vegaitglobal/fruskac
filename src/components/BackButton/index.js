// @flow

import { connect } from "react-redux";
import BackButton from "./BackButton";
import { onNavigateBack } from "../../actions/navigation";

const mapDispatchToProps = {
  onNavigateBack
};
const mapStateToProps = state => ({});

export default connect(mapStateToProps, mapDispatchToProps)(BackButton);
