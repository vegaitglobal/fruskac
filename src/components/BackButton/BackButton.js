// @flow
import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { TouchableOpacity, Text } from "react-native";
import I18n from "react-native-i18n";
import Icon from "react-native-vector-icons/Ionicons";

import Styles from "./Styles";

type Props = {
  onNavigateBack: () => void
};
type State = {};

class BackButton extends PureComponent<Props, State> {
  render() {
    const { onNavigateBack } = this.props;
    return (
      <TouchableOpacity style={Styles.buttonHolder} onPress={onNavigateBack}>
        <Icon name="ios-arrow-back" style={Styles.icon} size={30} />
        <Text style={Styles.text}>{I18n.t("back")}</Text>
      </TouchableOpacity>
    );
  }
}

export default BackButton;
