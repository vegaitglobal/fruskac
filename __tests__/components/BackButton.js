import "react-native";
import React from "react";
import BackButton from "../../src/components/BackButton/BackButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<BackButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
