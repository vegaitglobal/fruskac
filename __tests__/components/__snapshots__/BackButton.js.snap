// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginLeft": 8,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 30,
        },
        Object {
          "color": "#0066ff",
          "marginRight": 8,
          "top": 2,
        },
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#0066ff",
        "fontSize": 16,
      }
    }
  >
    [missing "en.back" translation]
  </Text>
</View>
`;
